libobs/media-io/video-io.c

```diff
@@ -271,12 +271,23 @@ void video_output_close(video_t *video)
 
 	video_output_stop(video);
 
-	for (size_t i = 0; i < video->inputs.num; i++)
-		video_input_free(&video->inputs.array[i]);
-	da_free(video->inputs);
+	if (video->initialized) {
+		video->initialized = false;
+		pthread_mutex_lock(&video->input_mutex);
 
-	for (size_t i = 0; i < video->info.cache_size; i++)
-		video_frame_free((struct video_frame *)&video->cache[i]);
+		for (size_t i = 0; i < video->inputs.num; i++)
+			video_input_free(&video->inputs.array[i]);
+		da_free(video->inputs);
+
+		for (size_t i = 0; i < video->info.cache_size; i++)
```

Comment:
Is this a memory leak that was previously unhandled?

Reply:
no the diff just looks weird you can see we were freeing video->inputs and video->cache before and are still freeing them after. But the diff interpreted the if statement as rewriting the freeing of video->inputs.

Reply:
Oh ok yeah I see that now, sorry.